examples/fluids/qfunctions/shocktube.h

```diff
@@ -0,0 +1,510 @@
+    // -- Interp-to-Interp q_data
+    const CeedScalar wdetJ      =   q_data[0][i];
+    // -- Interp-to-Grad q_data
+    // ---- Inverse of change of coordinate matrix: X_i,j
+    // *INDENT-OFF*
+    const CeedScalar dXdx[3][3] = {{q_data[1][i],
+                                    q_data[2][i],
+                                    q_data[3][i]},
+                                   {q_data[4][i],
+                                    q_data[5][i],
+                                    q_data[6][i]},
+                                   {q_data[7][i],
+                                    q_data[8][i],
+                                    q_data[9][i]}
+                                  };
+    // dU/dx
+    CeedScalar du[3][3] = {{0}};
+    CeedScalar drhodx[3] = {0};
+    CeedScalar dEdx[3] = {0};
+    CeedScalar dUdx[3][3] = {{0}};
+    CeedScalar dXdxdXdxT[3][3] = {{0}};
+    for (int j=0; j<3; j++) {
+      for (int k=0; k<3; k++) {
+        du[j][k] = (dU[j][k] - drho[k]*u[j]) / rho;
+        drhodx[j] += drho[k] * dXdx[k][j];
+        dEdx[j] += dE[k] * dXdx[k][j];
+        for (int l=0; l<3; l++) {
+          dUdx[j][k] += dU[j][l] * dXdx[l][k];
+          dXdxdXdxT[j][k] += dXdx[j][l]*dXdx[k][l];  //dXdx_j,k * dXdx_k,j
+        }
+      }
+    }
+
+    // *INDENT-ON*
+    const CeedScalar
+    E_kinetic  = 0.5 * rho * (u[0]*u[0] + u[1]*u[1] + u[2]*u[2]),
+    E_internal = E - E_kinetic,
+    P          = E_internal * (gamma - 1); // P = pressure
+
+    // The Physics
+    // Zero v and dv so all future terms can safely sum into it
+    for (int j=0; j<5; j++) {
+      v[j][i] = 0;
+      for (int k=0; k<3; k++)
+        dv[k][j][i] = 0;
+    }
+
+    // -- Density
+    // ---- u rho
+    for (int j=0; j<3; j++)
+      dv[j][0][i]  += wdetJ*(rho*u[0]*dXdx[j][0] + rho*u[1]*dXdx[j][1] +
+                             rho*u[2]*dXdx[j][2]);
+    // -- Momentum
+    // ---- rho (u x u) + P I3
+    for (int j=0; j<3; j++)
+      for (int k=0; k<3; k++)
+        dv[k][j+1][i]  += wdetJ*((rho*u[j]*u[0] + (j==0?P:0))*dXdx[k][0] +
+                                 (rho*u[j]*u[1] + (j==1?P:0))*dXdx[k][1] +
+                                 (rho*u[j]*u[2] + (j==2?P:0))*dXdx[k][2]);
+    // -- Total Energy Density
+    // ---- (E + P) u
+    for (int j=0; j<3; j++)
+      dv[j][4][i]  += wdetJ * (E + P) * (u[0]*dXdx[j][0] + u[1]*dXdx[j][1] +
+                                         u[2]*dXdx[j][2]);
+
+    // -- YZB stabilization
+    if (context->yzb) {
+      CeedScalar drho_norm = 0.0;   // magnitude of the density gradient
+      CeedScalar j_vec[3] = {0.0};  // unit vector aligned with the density gradient
+      CeedScalar jdgn[3] = {0.0};   // j * grad(N)
+      CeedScalar h_shoc = 0.0;      // element lengthscale
+      CeedScalar u_cha = 0.0;       // characteristic velocity, set to acoustic speed
+      CeedScalar tau_shoc = 0.0;    // timescale
+      CeedScalar nu_shoc = 0.0;     // artificial diffusion
```

Comment:
why `shoc` instead of `shock` and why not `j_gradn`? The idea is that people less familiar with fluids should still be able to use this example to help them understand how to write complex physics in QFunctions.